### PR TITLE
Enable TLS on customer data plane

### PR DESCRIPTION
**Pull request by zhan-airplane**
Needs https://github.com/airplanedev/airport/pull/3967. Sets up an AWS certificate and does DNS verification

### CODE REVIEW

File: data_plane.tf [hunk 1]
```diff
@@ -131,6 +143,92 @@ resource "aws_alb_listener" "alb_external_http" {
...
+    create_before_destroy = true
+  }
+}
+
+resource "aws_acm_certificate_validation" "alb_external_certificate_validation" {
+  certificate_arn = aws_acm_certificate.alb_external_certificate[0].arn
+  count = var.self_hosted_data_plane ? 1 : 0
+}
+
+resource "aws_alb_listener" "alb_external_https" {
```
Comment (yolken-airplane) [inline data_plane.tf:215]: Now that we have this, can we update the port 80 listener to just redirect to https? I think that's a common approach. Alternatively, we can just remove the insecure listener completely.

File: data_plane.tf [hunk 2]
```diff
@@ -131,6 +143,92 @@ resource "aws_alb_listener" "alb_external_http" {
...
+  domain_name = "${var.zone_slug}.${var.team_id}.${var.data_plane_domain}"
+  validation_method = "DNS"
+  count = var.self_hosted_data_plane ? 1 : 0
+
+  lifecycle {
+    create_before_destroy = true
+  }
+}
+
+resource "aws_acm_certificate_validation" "alb_external_certificate_validation" {
```
Comment (yolken-airplane) [inline data_plane.tf:210]: Out of curiosity, how long does this take?

**Comment by yolken-airplane** [comment]
Looks good, just one comment on the old insecure listener.

File: data_plane.tf (hunk 2 above)
Comment (zhan-airplane) [reply data_plane.tf:210]: This is pretty fast (~10s maybe? definitely less than a minute every time I've run it)

Comment (yolken-airplane) [reply data_plane.tf:210]: Nice!

**Review by yolken-airplane — APPROVED** [review]

**Pull request merged**